datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -211,19 +323,239 @@ pub(crate) fn min_statistics<'a, I: Iterator<Item = Option<&'a ParquetStatistics
+            MaxBooleanStatsIterator::new(iterator).map(|x| x.copied()),
+        ))),
+        DataType::Int8 => Ok(Arc::new(Int8Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| {
+                x.and_then(|x| {
+                    if let Ok(v) = i8::try_from(*x) {
+                        Some(v)
+                    } else {
+                        None
+                    }
+                })
+            }),
+        ))),
+        DataType::Int16 => Ok(Arc::new(Int16Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| {
+                x.and_then(|x| {
+                    if let Ok(v) = i16::try_from(*x) {
+                        Some(v)
+                    } else {
+                        None
+                    }
+                })
+            }),
+        ))),
+        DataType::Int32 => Ok(Arc::new(Int32Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| x.copied()),
+        ))),
+        DataType::Int64 => Ok(Arc::new(Int64Array::from_iter(
+            MaxInt64StatsIterator::new(iterator).map(|x| x.copied()),
+        ))),
+        DataType::UInt8 => Ok(Arc::new(UInt8Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| {
+                x.and_then(|x| {
+                    if let Ok(v) = u8::try_from(*x) {
+                        Some(v)
+                    } else {
+                        None
+                    }
+                })
+            }),
+        ))),
+        DataType::UInt16 => Ok(Arc::new(UInt16Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| {
+                x.and_then(|x| {
+                    if let Ok(v) = u16::try_from(*x) {
+                        Some(v)
+                    } else {
+                        None
+                    }
+                })
+            }),
+        ))),
+        DataType::UInt32 => Ok(Arc::new(UInt32Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| x.map(|x| *x as u32)),
+        ))),
+        DataType::UInt64 => Ok(Arc::new(UInt64Array::from_iter(
+            MaxInt64StatsIterator::new(iterator).map(|x| x.map(|x| *x as u64)),
+        ))),
+        DataType::Float32 => Ok(Arc::new(Float32Array::from_iter(
+            MaxFloatStatsIterator::new(iterator).map(|x| x.copied()),
+        ))),
+        DataType::Float64 => Ok(Arc::new(Float64Array::from_iter(
+            MaxDoubleStatsIterator::new(iterator).map(|x| x.copied()),
+        ))),
+        DataType::Date32 => Ok(Arc::new(Date32Array::from_iter(
+            MaxInt32StatsIterator::new(iterator).map(|x| x.copied()),
+        ))),
+        DataType::Date64 => Ok(Arc::new(Date64Array::from_iter(
+            MaxInt32StatsIterator::new(iterator)
+                .map(|x| x.map(|x| i64::from(*x) * 24 * 60 * 60 * 1000)),
+        ))),
+        DataType::Timestamp(_, _) => Ok(Arc::new(Int64Array::from_iter(
```

Comment:
I would have expected this to be a Timestamp array rather than an Int64 array 🤔

Reply:
It is consistent with the existing code

Reply:
I created another PR to fix the timestamp statistic types. Let me know if you want it to be part of this PR or a following PR.

https://github.com/xinlifoobar/datafusion/pull/1/files